### PR TITLE
{cae}[foss/2017b,intel/2017b] OpenFOAM v1712

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)
with patch obtained from @olesenm

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2553.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ef214a6d171adc3f8cf51d719e727faa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2010.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/feb8cefb13c44c07f0bb17b871ac7ac5 for a full test report.

**Comment by olesenm** [comment]
Nice to see that it seems to be working

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v1712-cleanup.patch [hunk 1]
```diff
@@ -0,0 +1,615 @@
+# - Replace the OpenFOAM third-party libraries with EASYBUILD variables.
+# - Set install dir, compiler, etc using EASYBUILD
+#
+# <Mark.Olesen@esi-group.com>
```
Comment (wpoely86) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v1712-cleanup.patch]: please don't change but extend the copyright on patches. This one is more or less the same as the one we already have?

Comment (olesenm) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v1712-cleanup.patch]: Sure - I don't actually care either way. The first patch was just a patch and someone requested that I add comment and copyright. So this time I added comments and copyright, but apparently not the right ones.
If it helps achieve what is required, I'm happy enough to work with a template if you what to provide one. 
However, I think that this whole patching business is a waste of time for OpenFOAM if we could instead just generate the config file contents directly (as per spack)

Comment (boegel) [reply easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v1712-cleanup.patch]: @OleHolmNielsen I fully agree that the patching is a PITA, and that we should avoid it if possible. Let's follow up on this in #1357.

@wpoely86 Patch is fixed in 5f908e9

**Pull request merged**